Sources/Plasma/Apps/plClient/CMakeLists.txt

```diff
@@ -90,17 +89,106 @@ if(WIN32)
         win32/res/Dirt.ICO
         win32/res/headspin.ico
     )
+elseif(APPLE)
+    include_directories("../../PubUtilLib/plPipeline/GL/OSX")
+    set(plClient_SOURCES ${plClient_SOURCES}
+        OSX/main.mm
+        OSX/PLSKeyboardEventMonitor.mm
+        OSX/PLSView.mm
+        OSX/PLSLoginWindowController.mm
+        OSX/PLSPatcherWindowController.mm
+        OSX/PLSPatcher.mm
+        OSX/PLSServerStatus.mm
+    )
+    set(plClient_HEADERS ${plClient_HEADERS}
+        OSX/PLSKeyboardEventMonitor.h
+        OSX/PLSView.h
+        OSX/PLSLoginWindowController.h
+        OSX/PLSPatcherWindowController.h
+        OSX/PLSPatcher.h
+        OSX/PLSServerStatus.h
+    )
+    set(RESOURCES 
+        OSX/MainMenu.xib
+        OSX/PLSLoginWindowController.xib
+        OSX/PLSPatcherWindowController.xib
+        OSX/banner@2x.jpeg
+        win32/res/Dirt.ICO
+    )
+    #shaders need to be compiled as part of the app
+    #this could change in the future, but for now the Metal code expects the library to be compiled in the app
```

Comment:
Ideally these would be compiled in as part of the pfMetalPipeline project

Reply:
Yeah, there are some decisions to be made here... Xcode likes to compile the shaders as part of the app so they can be packaged as part of the app. That's also why the non-bundle version of this got discarded. Metal shaders get compiled as part of the app compile and then put into the bundle.

If you don't do that - the shaders go into a metallib file. And that metallib file can be built separately and then manually brought into your app. But supposedly you then need to manually load the file when you load shaders. And this isn't unsupported. Stuff like the Metal compiler for Windows builds metallibs separate from an application build.

That's to say - all this requires some manual work to factor shader compile out to a different stage. And if the metallib is named default.metallib maybe Apple will load it as the default library even if it wasn't part of app compile?

Reply:
Also - this is what is requiring the Xcode generator. CMake doesn't have build in rules for dealing with Metal shaders or metallib files. But - that could be fixed by augmenting CMake with custom build commands. I started down this path at one point but never finished. The Xcode generator has worked well so far, but I could see more flexibility being desired for compiles. (The Xcode generator also manages code signing for us - if you have an account with Apple for signing.)

Reply:
`pfMetalPipeline` should probably be an `OBJECT` library.